test/shared/src/main/scala/zio/test/ZIOSpecDefault.scala

```diff
@@ -5,7 +5,7 @@ import zio.test.environment._
 
 abstract class ZIOSpecDefault extends ZIOSpec[TestEnvironment] {
 
-  final val layer: ZLayer[TestEnvironment, Any, TestEnvironment] = ZLayer.environment
+  final val testLayer: ZLayer[TestEnvironment, Any, TestEnvironment] = ZLayer.environment
```

Comment:
Why both `layer` and `testLayer`?

Reply:
At least my thought was that we wanted users to be able to use the `TestEnvironment` in implementing their own layer, and so if the signature has a dependency on `ZEnv` in `ZIOApp` it would be inconsistent to override it with `TestEnvironment` here. But to your point above maybe we just make the input type `Any` and allow the user to provide any needed environment themselves.

Reply:
I think this is another reason to remove `ZEnv` from `ZIOApp`. Because we can see from this example that different types of applications will have their own types of requirements, so while the default apps / specs can provide default environments, if you want the God mode you should wire everything together yourself (or use helper functions to add things like ZEnv, test environment, etc.). 

(Also from an app composition perspective, polymorphism in layer output is fine while polymorphism in layer input is not fine; so since polymorphism cannot solve this problem it may be best to push it onto the user.)

Reply:
Will change.